File: classic-cs-problems/src/main/ruby/calculating_pi.rb
def caculate(n)
  numberator = 4.0
  denominator = 1.0
  operation = 1.0
  pi = 0.0

  (0..n).each do
    pi += operation * (numberator / denominator)
    denominator += 2
    operation *= -1.0 
  end
  pi
end

puts caculate(1000000)

File: classic-cs-problems/src/main/python/calculating_pi.py
def calculate_pi(n_terms: int) -> float:
    numberator: float = 4.0
    denominator: float = 1.0
    operation: float = 1.0
    pi: float = 0.0
    for _ in range(n_terms):
        pi += operation * (numberator / denominator)
        denominator += 2.0
        operation *= -1.0
    return pi


if __name__ == '__main__':
    print(calculate_pi(1000000))
